### PR TITLE
pin worker nodes to worker nodes

### PR DESCRIPTION
**Pull request by xh3b4sd**
Fixes https://github.com/giantswarm/kvm-operator/issues/172. See also https://github.com/giantswarm/kvm-operator/pull/170.

### CODE REVIEW

File: service/resource/deployment/worker_deployment.go [hunk 1]
```diff
@@ -54,6 +54,9 @@ func newWorkerDeployments(customObject kvmtpr.CustomObject) ([]*extensionsv1.Dep
 					Spec: apiv1.PodSpec{
 						Affinity:    newWorkerPodAfinity(customObject),
 						HostNetwork: true,
+						NodeSelector: map[string]string{
+							"role": WorkerID,
```
Comment (calvix) [inline service/resource/deployment/worker_deployment.go]: what is this? what is `WorkerID` stands for?

In my PR we pin workers to nodes with labels `role=worker`

**Comment by calvix** [comment]
Merging of this is still blocked by `lycan` which doesn't have worker machines yet (but they are on the way)

File: service/resource/deployment/worker_deployment.go (hunk 1 above)
Comment (xh3b4sd) [reply service/resource/deployment/worker_deployment.go]: `WorkerID` is just a constant with the value `worker`. See https://github.com/giantswarm/kvm-operator/blob/acc1ea5106690dc59ab73463013d3c7005fea0a6/service/resource/deployment/resource.go#L23.

Comment (calvix) [reply service/resource/deployment/worker_deployment.go]: ok

**Comment by xh3b4sd** [comment]
Don't know but this is blocked because lycan is not ready. See Vaclav's comment.

**Comment by calvix** [comment]
Yes we have https://github.com/giantswarm/hive/blob/master/template_snippets/cloudconfig/g8s_worker.yaml#L89

**Comment by xh3b4sd** [comment]
Would be cool to get informed as soon as this is the case. <3

**Comment by calvix** [comment]
I sent technical info to the leaseweb guys now we are waiting for them to put servers in our rack and configure them

**Comment by calvix** [comment]
@xh3b4sd you have green to merge this, lycan machines are isntalled

**Comment by xh3b4sd** [comment]
Successfully tested on lycan.

**Comment by calvix** [comment]
what is stopping for merge now ?

**Comment by xh3b4sd** [comment]
Being on call.

**Pull request merged**